src/test/java/com/github/paweladamski/httpclientmock/HttpClientMockBuilderTest.java

```diff
@@ -314,6 +321,97 @@ public void should_allow_different_host_then_default() throws IOException {
     assertThat(login, hasContent("login"));
     assertThat(google, hasContent("google"));
   }
+  
+  @Test
+  public void withFormParameter() throws IOException {
```

Comment:
Does this test really work? I have doubts about it because you forgot to add `UrlEncodedFormCondition` to class `Rule` and this is necessary to actually check condition.

Reply:
The `UrlEncodedFormCondition` object was being added to the `RuleBuilder#conditions` list, which is why it works.